typing/ctype.ml

```diff
@@ -3033,7 +3047,8 @@ and unify_labeled_list env labeled_tl1 labeled_tl2 =
 
 and unify_package uenv lvl1 pack1 lvl2 pack2 =
   match
-    compare_package (get_env uenv) (unify_list uenv) lvl1 pack1 lvl2 pack2
+    compare_package (get_env uenv) (unify_list_same_length uenv)
```

Comment:
Mixed feelings. To the code reader it's not clear why we can assert here that the two lists have the same length; in general, of course, package types don't need to have the same number of equations to be equivalent. The reason we can make this assumption is internal to `compare_package`: its implementation will complete the lists to have the same length. But then, why is the assertion stated here, instead of being made internally within `compare_package`, on the return of the completion machinery and before called the `unify_list` function?

Reply:
I disagree, `unify_list` cannot be used in `compare_package` because it raises a semantic error trace which does not happen in the `compare_package` case.

Thus `compare_package`  should call its own `unify_list` variant, and I prefer to have a variant that makes it clear that calling it on lists with different lengths is an error.

Reply:
Then I would propose that the function be parametrized over `unify` rather than `unify_list`, with the `List.iter2` call inlined in the caller.

Reply:
That sounds like the best option to me.